core/server/api_container/server/startosis_engine/kurtosis_types/service_config/service_config.go

```diff
@@ -211,68 +211,33 @@ func (config *ServiceConfig) Copy() (builtin_argument.KurtosisValueType, error)
-	}
-
-	// TODO: Sad that we need the service network here to get the APIC info. This is wrong, we should fix this by
-	//  passing the APIC info DOWN to the backend and have the backend create the expander itself.
-	//  Here writing those info into each service config is dumb
-	apiContainerInfo := serviceNetwork.GetApiContainerInfo()
-	filesArtifactsExpanderArgs, err := args.NewFilesArtifactsExpanderArgs(
-		apiContainerInfo.GetIpAddress().String(),
-		apiContainerInfo.GetGrpcPortNum(),
-		filesArtifactsExpansions,
-	)
-	if err != nil {
-		return nil, startosis_errors.NewInterpretationError("An error occurred creating files artifacts expander args")
-	}
-
-	expanderEnvVars, err := args.GetEnvFromArgs(filesArtifactsExpanderArgs)
-	if err != nil {
-		return nil, startosis_errors.NewInterpretationError("An error occurred getting files artifacts expander environment variables using args: %+v", filesArtifactsExpanderArgs)
-	}
-
-	expanderImageAndTag := fmt.Sprintf(
-		"%v:%v",
-		filesArtifactsExpanderImage,
-		apiContainerInfo.GetVersion(),
-	)
-
-	return &service_directory.FilesArtifactsExpansion{
-		ExpanderImage:                        expanderImageAndTag,
-		ExpanderEnvVars:                      expanderEnvVars,
-		ServiceDirpathsToArtifactIdentifiers: serviceDirpathsToArtifactIdentifiers,
-		ExpanderDirpathsToServiceDirpaths:    expanderDirpathToUserServiceDirpathMap,
-	}, nil
-}
-
-func convertPersistentDirectoryMounts(persistentDirectoriesMap map[string]service_directory.PersistentDirectory) *service_directory.PersistentDirectories {
-	return service_directory.NewPersistentDirectories(persistentDirectoriesMap)
-}
-
-func (config *ServiceConfig) ToKurtosisType(serviceNetwork service_network.ServiceNetwork) (*service.ServiceConfig, *startosis_errors.InterpretationError) {
+func (config *ServiceConfig) ToKurtosisType(
+	serviceNetwork service_network.ServiceNetwork,
+	locatorOfModuleInWhichThisBuiltInIsBeingCalled string,
+	packageId string,
+	packageContentProvider startosis_packages.PackageContentProvider,
+	packageReplaceOptions map[string]string,
+) (*service.ServiceConfig, *startosis_errors.InterpretationError) {
 	var ok bool
-	image, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[starlark.String](config.KurtosisValueTypeDefault, ImageAttr)
+
+	var imageName string
+	var imageBuildSpec *image_build_spec.ImageBuildSpec
+	rawImageAttrValue, found, interpretationErr := kurtosis_type_constructor.ExtractAttrValue[starlark.Value](config.KurtosisValueTypeDefault, ImageAttr)
 	if interpretationErr != nil {
 		return nil, interpretationErr
 	}
 	if !found {
-		return nil, startosis_errors.NewInterpretationError("Required attribute '%s' could not be found on type '%s'",
-			ImageAttr, ServiceConfigTypeName)
+		return nil, startosis_errors.NewInterpretationError("Required attribute '%s' could not be found on type '%s'", ImageAttr, ServiceConfigTypeName)
+	}
+	imageName, imageBuildSpec, interpretationErr = convertImage(
```

Comment:
nit `imageBuildSpec` could be renamed to `maybeImageBuildSpec`

Reply:
good idea